src/renderer/pages/vpn.vue

```diff
@@ -158,12 +171,40 @@ export default {
     onCountriesUpdate (countries) {
       this.countriesAreLoading = false
 
-      if (countries.length < 1) this.bugReporter.captureInfoMessage('Renderer received empty countries list')
+      if (countries.length < 1) {
+        this.bugReporter.captureInfoMessage('Renderer received empty countries list')
+      }
 
       this.countryList = countries
+    },
+    onTabClick (page) {
+      if (page !== 'provider') {
+        return
+      }
+
+      if (this.statusCode > -1) {
+        this.showTabModal = true
+        return
+      }
+
+      this.goToServicePage()
+    },
+    toServicePage () {
+      if (this.statusCode > -1) {
+        this.$store.dispatch(type.DISCONNECT)
```

Comment:
Shouldn't we await here?
```suggestion
        await this.$store.dispatch(type.DISCONNECT)
```

Reply:
changed